controllers/doguInstallManager.go

```diff
@@ -0,0 +1,429 @@
+	}
+
+	imageRegistry := registry.NewCraneContainerImageRegistry(operatorConfig.DockerRegistry.Username, operatorConfig.DockerRegistry.Password)
+	resourceGenerator := resource.NewResourceGenerator(client.Scheme())
+	restConfig := ctrl.GetConfigOrDie()
+	clientSet, err := kubernetes.NewForConfig(restConfig)
+	if err != nil {
+		return nil, fmt.Errorf("failed find cluster config: %w", err)
+	}
+
+	fileExtract := newPodFileExtractor(client, restConfig, clientSet)
+	applier, scheme, err := apply.New(restConfig, k8sDoguOperatorFieldManagerName)
+	if err != nil {
+		return nil, fmt.Errorf("failed create K8s Applier: %w", err)
+	}
+	err = k8sv1.AddToScheme(scheme)
+	if err != nil {
+		return nil, fmt.Errorf("failed add applier scheme to dogu CRD scheme handling: %w", err)
+	}
+
+	doguRegistrator := NewCESDoguRegistrator(client, cesRegistry, resourceGenerator)
+	dependencyValidator := dependency.NewCompositeDependencyValidator(operatorConfig.Version, cesRegistry.DoguRegistry())
+
+	executor := resource.NewCommandExecutor(clientSet, clientSet.CoreV1().RESTClient())
+	serviceAccountCreator := serviceaccount.NewCreator(cesRegistry, executor)
+
+	return &doguInstallManager{
+		Client:                client,
+		Scheme:                client.Scheme(),
+		ResourceGenerator:     resourceGenerator,
+		DoguRemoteRegistry:    doguRemoteRegistry,
+		DoguLocalRegistry:     cesRegistry.DoguRegistry(),
+		ImageRegistry:         imageRegistry,
+		DoguRegistrator:       doguRegistrator,
+		DependencyValidator:   dependencyValidator,
+		ServiceAccountCreator: serviceAccountCreator,
+		DoguSecretHandler:     resource.NewDoguSecretsWriter(client, cesRegistry),
+		FileExtractor:         fileExtract,
+		Applier:               applier,
+	}, nil
+}
+
+// Install installs a given Dogu Resource. This includes fetching the dogu.json and the container image. With the
+// information Install creates a Deployment and a Service
+func (m *doguInstallManager) Install(ctx context.Context, doguResource *k8sv1.Dogu) error {
+	logger := log.FromContext(ctx)
+
+	doguResource.Status = k8sv1.DoguStatus{RequeueTime: doguResource.Status.RequeueTime, Status: k8sv1.DoguStatusInstalling, StatusMessages: []string{}}
+	err := doguResource.Update(ctx, m.Client)
+	if err != nil {
+		return fmt.Errorf("failed to update dogu status: %w", err)
+	}
+
+	// Set the finalizer at the beginning of the install procedure.
+	// This is required because an error during installation would leave a dogu resource with its
+	// k8s resources in the cluster. A delete would tidy up those resources but would not start the
+	// delete procedure from the controller.
+	logger.Info("Add dogu finalizer...")
+	controllerutil.AddFinalizer(doguResource, finalizerName)
+	err = m.Client.Update(ctx, doguResource)
+	if err != nil {
+		return fmt.Errorf("failed to update dogu: %w", err)
+	}
+
+	// we need to retrieve the config map with the custom descriptor to delete it after ending the installation
```

Comment:
Puhh.. first of all the config map is created by an user. To be more precise from a dev. On the one side it sounds not bad to delete it automatically. On the other side it would be nice if the configmap still exists if you install for example nginx multiple times in a development phase. I am unsure too :smile: